### DISCUSSION TITLE
How to create a dataset of adjoint optimised silicon photonics devices fast?

### DISCUSSION THREAD

**Discussion by satadrudas** [NanoComp/meep#2553] [Q&A] [answered]
This may be a naive question so I apologise in advance.

I am currently working on a project on silicon photonics inverse designs. For now I am trying to create a dataset of the adjoint optimised devices (take for example a WDM ) from different initialising conditions (filters, minimum length, initial permittivity etc) however each optimisation is frustratingly slow and I need at least 500-1000 different designs.

Is there any way to accelerate this process or any way such that meep spits out several designs together? (they don't have to be fully optimised).

Thanks

**Comment by smartalecH** [comment, marked as answer]
Keep in mind that the adjoint solver simply computes a gradient. That's it. And only a single gradient (for each wavelength you specify).

If you want to design multiple devices in tandem, you need to run multiple optimization in tandem, which means you need to simulate multiple devices in tandem. There's no real way around this...

You can always just throw more hardware at it if you want to design more devices more quickly.

**Reply by satadrudas** [reply]
Alright. Thanks
